### PR TITLE
Develop hp timeline

### PR DESCRIPTION
**Pull request by ben2collins**

### CODE REVIEW

**Comment by ben2collins** [comment]
@drogers98  PR for Timeline Widget is ready

File: modules/hp_timeline/templates/paragraph--hp-timeline-field.html.twig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<div class="entry">
+    <div class="title">
+        <h3>{{ content.field_hp_timeline_field_datetime[0] }}</h3>
+            {% if content.field_hp_timeline_field_title[0] is not null %}
```
Comment (drogers98) [inline modules/hp_timeline/templates/paragraph--hp-timeline-field.html.twig]: remove indent for this block, should be same indent level as h3

File: modules/hp_timeline/templates/paragraph--hp-timeline-field.html.twig [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+                <p>{{ content.field_hp_timeline_field_title[0] }}</p>
```
Comment (drogers98) [inline modules/hp_timeline/templates/paragraph--hp-timeline-field.html.twig]: is the 0 needed here? why isnt it just {{ {{ content.field_hp_timeline_field_title }} ?

File: modules/hp_timeline/templates/paragraph--hp-timeline-field.html.twig [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+            {% endif %}
+    </div>
+    <div class="body">
+        {% if content.field_hp_timeline_field_desc[0] is not null %}
+            <p>{{ content.field_hp_timeline_field_desc[0] }}</p>
```
Comment (drogers98) [inline modules/hp_timeline/templates/paragraph--hp-timeline-field.html.twig]: same 0 comment

File: modules/hp_timeline/templates/paragraph--hp-timeline.html.twig [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<section class="section--padded-top section--padded-bottom"  aria-label="Timeline">
+    {% if hp_show_wrapper %}
+        <section class="compartment">
+    {% endif %}
+            <div>
+                {% if content.field_timeline_title is not null %}
+                    <p class="lead-in">{{ content.field_timeline_title[0] }}</p>
```
Comment (drogers98) [inline modules/hp_timeline/templates/paragraph--hp-timeline.html.twig]: same 0 comment

File: modules/hp_timeline/templates/paragraph--hp-timeline.html.twig [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+                {% endif %}
+                {% if content.field_timeline_text is not null %}
+                    <p>{{ content.field_timeline_text[0] }}</p>
```
Comment (drogers98) [inline modules/hp_timeline/templates/paragraph--hp-timeline.html.twig]: same 0 comment

File: modules/hp_timeline/templates/paragraph--hp-timeline.html.twig [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+                {% endif %}
+            </div>
+            <div class="timeline">
+                {% for i in 0..content.field_hp_timeline_field|length %}
```
Comment (drogers98) [inline modules/hp_timeline/templates/paragraph--hp-timeline.html.twig]: This loop seems odd. I get what you are doing, but using 0..content.whatever seems off. Is there a cleaner way to get each item in this array?

File: modules/hp_timeline/config/install/field.storage.paragraph.field_timeline_text.yml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+langcode: en
+status: true
+dependencies:
+  module:
+    - paragraphs
+id: paragraph.field_timeline_text
```
Comment (drogers98) [inline modules/hp_timeline/config/install/field.storage.paragraph.field_timeline_text.yml]: should this be field_hp_timeline_text? Remember we want to prefix hp wherever possible to indicate that this field is controlled by howard paragraphs module.

File: modules/hp_timeline/config/install/field.storage.paragraph.field_timeline_title.yml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+langcode: en
+status: true
+dependencies:
+  module:
+    - paragraphs
+id: paragraph.field_timeline_title
+field_name: field_timeline_title
```
Comment (drogers98) [inline modules/hp_timeline/config/install/field.storage.paragraph.field_timeline_title.yml]: should this be field_hp_timeline_text? Remember we want to prefix hp wherever possible to indicate that this field is controlled by howard paragraphs module.

File: modules/hp_timeline/config/install/field.field.paragraph.hp_timeline.field_timeline_title.yml [hunk 9]
```diff
@@ -0,0 +1,18 @@
+langcode: en
+status: true
+dependencies:
+  config:
+    - field.storage.paragraph.field_timeline_title
+    - paragraphs.paragraphs_type.hp_timeline
+id: paragraph.hp_timeline.field_timeline_title
+field_name: field_timeline_title
```
Comment (drogers98) [inline modules/hp_timeline/config/install/field.field.paragraph.hp_timeline.field_timeline_title.yml]: should this be field_hp_timeline_text? Remember we want to prefix hp wherever possible to indicate that this field is controlled by howard paragraphs module.

File: modules/hp_timeline/config/install/field.field.paragraph.hp_timeline.field_timeline_text.yml [hunk 10]
```diff
@@ -0,0 +1,18 @@
+langcode: en
+status: true
+dependencies:
+  config:
+    - field.storage.paragraph.field_timeline_text
+    - paragraphs.paragraphs_type.hp_timeline
+id: paragraph.hp_timeline.field_timeline_text
+field_name: field_timeline_text
```
Comment (drogers98) [inline modules/hp_timeline/config/install/field.field.paragraph.hp_timeline.field_timeline_text.yml]: should this be field_hp_timeline_text? Remember we want to prefix hp wherever possible to indicate that this field is controlled by howard paragraphs module.

**Review by drogers98 — COMMENTED** [review]
I did not pull down and test, but have some comments added. Additionally, the module should be added to the list in the readme of the main module, where it lists submodules.

**Comment by ben2collins** [comment]
@drogers98 I applied all of your recommended changes to my to latest commit.

**Pull request merged**